aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -727,6 +769,8 @@ public AwsProperties() {
 
     this.s3SignerImpl = null;
     this.allProperties = Maps.newHashMap();
+    this.clientRegion = null;
```

Comment:
nit: all `clientXXX` variables are placed in the block on L682, I think we should move these variables also there

Reply:
moved

Reply:
Sorry seems like these variables are not moved yet?

Reply:
hi @jackye1995 I think i moved, but missed to group. Please check again. thanks

Reply:
I think it's still not moved? Maybe we are talking across each other. What I meant was:

```
this.clientRegion = null;
this.clientCredentialsProvider = null;
this.clientCredentialsProviderProperties = null;
```

can be moved next to 

```
this.clientAssumeRoleArn = null;
this.clientAssumeRoleTimeoutSec = CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT;
this.clientAssumeRoleExternalId = null;
this.clientAssumeRoleRegion = null;
this.clientAssumeRoleSessionName = null;
```
, and similarly

```
this.clientRegion = properties.get(CLIENT_REGION);
this.clientCredentialsProvider = properties.get(CLIENT_CREDENTIALS_PROVIDER);
this.clientCredentialsProviderProperties = PropertyUtil.propertiesWithPrefix(properties, CREDENTIAL_PROVIDER_PREFIX);
```

and 

```
private String clientRegion;
private String clientCredentialsProvider;
private final Map<String, String> clientCredentialsProviderProperties;
```

can be moved to the corresponding places where existing `clientXXX` variables are.